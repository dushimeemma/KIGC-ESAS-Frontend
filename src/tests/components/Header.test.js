import React from 'react';
import { shallow } from 'enzyme';
import { useDispatch } from 'react-redux';

import Navs from '../../components/Header';

const mockDispatch = jest.fn();
const onClick = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('<Navs/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Navs />);
  });
  it('Should render navigation', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should logout user', () => {
    let username = 'dushime emmanuel';
    localStorage.setItem('username', username);
    const wrapper = shallow(<Navs />);
    const logout = wrapper.find('#logout');
    logout.simulate('click', { onClick });
    expect(wrapper).toMatchSnapshot();
  });
  afterEach(() => {
    localStorage.removeItem('username');
  });
});
