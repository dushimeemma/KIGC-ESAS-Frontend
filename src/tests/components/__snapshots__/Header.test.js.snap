// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navs/> Should logout user 1`] = `
<Navbar
  className="back-color mb-2"
  expand={false}
  tag="nav"
>
  <NavbarBrand
    className="text-color"
    href="/"
    tag="a"
  >
    KIGC | ESAS
  </NavbarBrand>
  <Nav
    className="ml-auto"
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="text-color"
        href="/dashboard"
        tag="a"
      >
        Welcome 
        <strong
          className="text-capitalize"
        >
          dushime emmanuel
        </strong>
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className="text-color cursor-pointer"
        id="logout"
        onClick={[Function]}
        tag="a"
      >
        Logout
      </NavLink>
    </NavItem>
  </Nav>
</Navbar>
`;

exports[`<Navs/> Should render navigation 1`] = `
<Navbar
  className="back-color mb-2"
  expand={false}
  tag="nav"
>
  <NavbarBrand
    className="text-color"
    href="/"
    tag="a"
  >
    KIGC | ESAS
  </NavbarBrand>
  <Nav
    className="ml-auto"
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="text-color"
        href="/login"
        tag="a"
      >
        Login
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className="text-color"
        href="/register"
        tag="a"
      >
        Register
      </NavLink>
    </NavItem>
  </Nav>
</Navbar>
`;
